pre_commit_hooks/insert_license.py

```diff
@@ -196,9 +209,15 @@ def license_not_found(remove_header, license_info, src_file_content, src_filepat
         index = 0
         for line in src_file_content:
             stripped_line = line.strip()
-            # Special treatment for shebang, encoding and empty lines when at the beginning of the file
+            # Special treatment for user provided regex,
+            # or shebang, file encoding directive,
+            # and empty lines when at the beginning of the file.
             # (adds license only after those)
-            if stripped_line.startswith("#!") \
+            if after_regex is not None and after_regex != "":
```

Comment:
```suggestion
            if after_regex:
```

This would cover both cases of `None` and the empty string

Reply:
`If after_regex:` is not the same.

Technically "if 0:" is also false as well as several other expressions.

This way of coding explicitly shows that it's testing for None and empty strings which conveys a meaning to the reader.

Shorter is not always better if it's not explicit. Or one has to add a comment.

I vaguely remember a case where the code had 'if x:' and it did not work because it matched more (or less) than the coder expected..

It would be better to write
`if  isinstance(after_regex, str) and after_regex:`

`if after_regex:` does not fit my code quality requirements as there is no reason to make code as short as possible here.

But I am not enforcing my style ;-).

Reply:
> But I am not enforcing my style ;-).

I think you are a little bit, but I'm fine with it 😃

Reply:
> I think you are a little bit

I am still allowed to argue about it ;-) and get (some level of) your agreement.

I can do short, optimized, one-liners, etc.  One of my best optimisations was to rewrite a c-code that took 12 minutes to execute down to 1min 20sec to execute.  I didn't change the algorithm that much, but "reworded" the code with compiler optimisation and ultimate machine code in mind.

But there is a tradeoff between avoiding future mistakes, speed, size, readability, etc.

Reply:
If you're into optimizations, you may be interested in this article:
https://mazzo.li/posts/fast-pipes.html

Au fait je crois qu'on est français tous les deux ^^
Dans ce contexte précis, le typage de `after_regex` était assuré par `argparse`, donc il y avait peu de chances qu'une valeur de `0` advienne ! Mais je comprends la mentalité / approche globale de programmation défensive, et je n'ai pas d'objection sur le fond 😊